internal/controller/biossettings_controller.go

```diff
@@ -757,45 +1093,43 @@ func (r *BiosSettingsReconciler) updateBiosSettingsStatus(
 	log logr.Logger,
 	biosSettings *metalv1alpha1.BIOSSettings,
 	state metalv1alpha1.BIOSSettingsState,
+	condition *metav1.Condition,
 ) error {
-	if biosSettings.Status.State == state {
+	if biosSettings.Status.State == state && condition == nil {
 		return nil
 	}
 
 	biosSettingsBase := biosSettings.DeepCopy()
 	biosSettings.Status.State = state
 
 	if state == metalv1alpha1.BIOSSettingsStateApplied {
-		biosSettings.Status.UpdateSettingState = ""
+		time := metav1.Now()
+		biosSettings.Status.AppliedStateTimeStamp = &time
+	} else if !biosSettings.Status.AppliedStateTimeStamp.IsZero() {
+		biosSettings.Status.AppliedStateTimeStamp = &metav1.Time{}
 	}
 
-	if err := r.Status().Patch(ctx, biosSettings, client.MergeFrom(biosSettingsBase)); err != nil {
-		return fmt.Errorf("failed to patch BIOSSettings status: %w", err)
-	}
-
-	log.V(1).Info("Updated biosSettings state ", "new state", state)
-
-	return nil
-}
-
-func (r *BiosSettingsReconciler) updateBIOSSettingUpdateStatus(
-	ctx context.Context,
-	log logr.Logger,
-	biosSettings *metalv1alpha1.BIOSSettings,
-	state metalv1alpha1.BIOSSettingUpdateState,
-) error {
-	if biosSettings.Status.UpdateSettingState == state {
-		return nil
+	if condition != nil {
+		acc := conditionutils.NewAccessor(conditionutils.AccessorOptions{})
+		if err := acc.UpdateSlice(
+			&biosSettings.Status.Conditions,
+			condition.Type,
+			conditionutils.UpdateStatus(condition.Status),
+			conditionutils.UpdateReason(condition.Reason),
+			conditionutils.UpdateMessage(condition.Message),
+		); err != nil {
+			return fmt.Errorf("failed to patch BIOSettings condition: %w", err)
+		}
+	} else if state == metalv1alpha1.BIOSSettingsStatePending {
+		// reset, when we restart the setting update
+		biosSettings.Status.Conditions = nil
 	}
 
-	biosSettingsBase := biosSettings.DeepCopy()
-	biosSettings.Status.UpdateSettingState = state
-
 	if err := r.Status().Patch(ctx, biosSettings, client.MergeFrom(biosSettingsBase)); err != nil {
-		return fmt.Errorf("failed to patch BIOSSettings UpdateSetting status: %w", err)
+		return fmt.Errorf("failed to patch BIOSSettings status: %w", err)
 	}
 
-	log.V(1).Info("Updated biosSettings UpdateSetting state ", "new state", state)
+	log.V(1).Info("Updated biosSettings state ", "new state", state, "new conditions", biosSettings.Status.Conditions)
```

Comment:
We should use here something like `"State", state` in the info key value map. That way it is easier to query for that in the logging backend.